File: scala/src/main/scala/com/vdzon/administratie/checkandfix/actions/fix/BoekingenFix.scala
package com.vdzon.administratie.checkandfix.actions.fix

import com.vdzon.administratie.checkandfix.model.{CheckAndFixRegel, FixAction}
import com.vdzon.administratie.model.Administratie

import scala.collection.JavaConversions._
import scala.compat.java8.StreamConverters._

object BoekingenFix {

  def removeBoekingen(regel: CheckAndFixRegel, administratie: Administratie): Administratie = {
    if (regel.rubriceerAction ne FixAction.REMOVE_BOEKING) return administratie
    val nieuweBoekingen = administratie.getBoekingen
      .stream
      .toScala[Stream]
      .filter(boeking => boeking.getUuid.ne(regel.boekingUuid))

    return Administratie.newBuilder(administratie).boekingen(nieuweBoekingen).build
  }


}

File: checkandfix/impl/src/main/java/com/vdzon/administratie/checkandfix/actions/fix/BoekingenFix.kt
package com.vdzon.administratie.checkandfix.actions.fix

import com.vdzon.administratie.checkandfix.CheckAndFixRegel
import com.vdzon.administratie.checkandfix.model.FixAction
import com.vdzon.administratie.model.Administratie
import com.vdzon.administratie.model.boekingen.Boeking

object BoekingenFix {

    fun removeBoekingen(regel: CheckAndFixRegel, administratie: Administratie): Administratie {
        if (regel.rubriceerAction != FixAction.REMOVE_BOEKING) return administratie
        val nieuweBoekingen: List<Boeking> = administratie.boekingen.toMutableList().filter { boeking -> boeking.uuid != regel.boekingUuid }
        return administratie.copy(boekingen = nieuweBoekingen.toMutableList())
    }
}
